mrgeo-core/src/main/scala/org/mrgeo/job/JobArguments.scala

```diff
@@ -362,4 +368,131 @@ class JobArguments() extends Logging {
     }
   }
 
+  def loadYarnSettings() = {
+
+    val res = calculateYarnResources()
+    val sparkConf = SparkUtils.getConfiguration
+
+    val minmemory = sparkConf.getLong(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB,
+      YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB)
+    val maxmemory = sparkConf.getLong(YarnConfiguration.RM_SCHEDULER_MAXIMUM_ALLOCATION_MB,
+      YarnConfiguration.DEFAULT_RM_SCHEDULER_MAXIMUM_ALLOCATION_MB)
+
+    val executorMemoryOverhead = sparkConf.getInt("spark.yarn.executor.memoryOverhead", 384)
+    val mem = res._3
+    val actualoverhead = ((if ((mem * YarnSparkHadoopUtil.MEMORY_OVERHEAD_FACTOR) > executorMemoryOverhead) mem * YarnSparkHadoopUtil.MEMORY_OVERHEAD_FACTOR
+    else executorMemoryOverhead) * 0.95).toLong
+
+    cores = res._1
+    executors = res._2 // reserve 1 executor for the driver
+    executorMemKb = (mem - actualoverhead) * 1024 // memory per worker
+    memoryKb = res._4 * 1024  // total memory (includes driver memory)
+
+    logInfo("Configuring job (" + name + ") with " + (executors + 1) + " workers (1 driver, " + executors + " executors)  with " +
+        cores + " threads each and " + SparkUtils.kbtohuman(memoryKb, "m") +
+        " total memory, " + SparkUtils.kbtohuman(executorMemKb + (actualoverhead * 1024), "m") +
+        " per worker (" + SparkUtils.kbtohuman(executorMemKb, "m") + " + " +
+        SparkUtils.kbtohuman(actualoverhead * 1024, "m") + " overhead per task)" )
+  }
+
+
+  private def configureYarnMemory(cores:Int, nodes:Int, memory:Long, unitMemory:Long, minMemory:Long, maxMemory:Long) = {
+
+    val rawMemoryPerNode = memory / nodes
+    val rawExecutorsPerNode = cores / nodes
+    val rawMemPerExecutor = rawMemoryPerNode / rawExecutorsPerNode
+
+    val rawUnits = Math.floor(rawMemPerExecutor.toDouble / unitMemory)
+
+    val units = {
+      val r = rawUnits * unitMemory
+      if (r > maxMemory)
+      // Make this is a multiple of unitMemory
+        Math.floor(maxMemory.toDouble / unitMemory.toDouble).toInt
+      else if (r < minMemory)
+      // Make this is a multiple of unitMemory
+        Math.ceil(minMemory.toDouble / unitMemory.toDouble).toInt
+      else
+        rawUnits
+    }
+
+    val executorMemory = units * unitMemory
+
+    val executorsPerNode = (rawMemoryPerNode.toDouble / executorMemory).toInt
+    val executors = executorsPerNode * nodes
+
+    (executors.toInt, executorMemory.toLong)
+  }
+
+  private def calculateYarnResources():(Int, Int, Long, Long) = {
+
+    val cl = getClass.getClassLoader
+
+    val client:Class[_] =
+      try {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://www.codacy.com/app/DGIS/mrgeo/file/2578646992/issues/source?bid=3293399&fileBranchId=3301986#l432)